### PR TITLE
x-auth request header use changes

### PR DESCRIPTION
**Pull request by vishal22**

### CODE REVIEW

**Comment by leadingcode** [comment]
Good idea!

**Comment by coolbung** [comment]
@vishal22 this line should be removed. The authentication class should be simply decided based on the header value. The default value of this should be picked up from the config.

So in the contructor you should do

$auth_method = headers['x-auth'] ? headers['x-auth'] : $params->get('auth_method');
$this->set('auth_method', $auth_method);

Then remove all the code related to key/login checks. All that code is unnecessary. Great chance for #strengthbysubtraction !

In the authenticateRequest() method just have

$className 		= 'APIAuthentication'.ucwords($this->get('auth_method'));

**Comment by coolbung** [comment]
@vishal22 you should use http://php.net/manual/en/function.getallheaders.php instead of the apache specific function

**Comment by coolbung** [comment]
@vishal22 knock knock..

File: code/site/libraries/authentication.php [hunk 1]
```diff
@@ -50,15 +57,28 @@ public static function authenticateRequest() {
...
 		else
-		$method			= 'login';
+		{
+			$method			= 'login';
+		}
+		
+		if($head_auth)
+		{
+			$method			= 'session';
+		}
```
Comment (coolbung) [inline code/site/libraries/authentication.php]: @vishal22 Why is all this code present here ?

**Pull request merged**